app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/localBackend.ts

```diff
@@ -414,4 +412,24 @@ export class LocalBackend extends backend.Backend {
     override deleteTag() {
         return Promise.resolve()
     }
+
+    // === Protected methods ===
+
+    /** Get a fallback engine version. */
+    protected async getFallbackEngineVersion(): Promise<string> {
+        if (this.fallbackEngineVersion != null) {
+            return this.fallbackEngineVersion
+        } else {
+            const versions = await this.listVersions({
+                versionType: backend.VersionType.backend,
+                default: true,
+            })
+            const version = versions[versions.length - 1]
```

Comment:
What if the `versions` array is zero length?

Reply:
then it'll be out of bounds so it'll be `undefined` - the condition below checks for that case, and throws if `version` is `undefined` (meaning `versions` was zero length)

Reply:
Ááá, JavaScript:
```
> undefined == null
true
```